### ISSUE TITLE
"Password is incorrect" when AuthPro generates a "corrupted" export file

### ISSUE THREAD

**Issue by Nowaker** [jamie-mh/AuthenticatorPro#1165]
**Describe the bug**

When two entries with the same issuer and username but a different secret exist, AuthPro will generate a JSON backup file that it will refuse to import claiming the password is incorrect.

Related to https://github.com/jamie-mh/AuthenticatorPro/issues/119

**To Reproduce**
1. Get AuthPro to a state where it has two secrets for the same account.
2. Export.
3. Import.
4. Error. 

**Expected behavior**

1. Auth Pro generates an export that AuthPro cannot export (crazy, right? 😆)
  - If it's a valid state in my current AuthPro, it should be a valid state in my new AuthPro. So just import it as-is.
2. Fix non-factual and confusing error message "Password is incorrect" when in reality, something else is an issue.
  - If the export is password-protected, it should not say "Password is incorrect" since there is no password.
  - If the export is password-protected, and the supplied password is correct, it should not say "Password is incorrect" since it's fine.
  - Errors related to "corruption" / failure to import should be messaged in a different way.
3. Validate an export file after it's generated.
  - The same function that validates if an export is "corrupted" during import should be proactively run after an export completes to validate the format, and inform the user that they may encounter a problem during the import due to an unknown bug.

**App Version**: 1.25.2

**Additional context**

JSON file had these two entries. EMAIL and ABC and DEF are snipped.

```json
{
    {
      "Type": 2,
      "Icon": "stripe",
      "Issuer": "Stripe",
      "Username": "EMAIL",
      "Secret": "ABC",
      "Pin": null,
      "Algorithm": 0,
      "Digits": 6,
      "Period": 30,
      "Counter": 0,
      "CopyCount": 0,
      "Ranking": 19
    },
    {
      "Type": 2,
      "Icon": "stripe",
      "Issuer": "Stripe",
      "Username": "EMAIL",
      "Secret": "DEF",
      "Pin": null,
      "Algorithm": 0,
      "Digits": 6,
      "Period": 30,
      "Counter": 0,
      "CopyCount": 0,
      "Ranking": 20
    }
}
```

**Comment by Nowaker** [comment]
Oh shit...

I was selecting "Import from other apps", and then choosing "Authenticator Plus".

The UX is to blame.

"Import from other apps" wasn't the first place where I'd look to import... After all, it's called "Import from other apps", so I was clicking around, and around, and couldn't find the import UI. So I clicked "Import from other apps", and there it was Authenticator Pro, or so I thought! I even joked about it to my wife.

The core UX problem is the import feature present under the  `[ + ]` button. As a long-time user of AuthPro, I've used this button tens of times - always for one purpose - adding a new QR codes to the application. **It's not a place where I'd look for import from backup / other apps feature.** Especially when the home screen says "Nothing here", gives a "Getting started guide" and "Import from other apps".

To avoid confusion, the second option on an empty screen should be "Import from Authenticator Pro backup", and the third "Import from other apps". Or, rename "Import from other apps" to "Import from other apps or backup", with Authenticator Pro being listed aside other third party apps on the list.

The original report partially stands: error handling and messaging needs to be improved. But the "duplicate" entry wasn't a problem - I read about duplicates being at issue previously, so I assumed that was the case.

**Comment by Red6785** [comment]
I agree, the UX for importing backups is more than a bit confusing. I think for ease of organization/understanding, you should rename your issue to reflect the actual issue.

**Comment by d3vnate** [comment]
Ugh. Yes, annoying for sure.  Thanks to this post, helped alleviate a fair amount of frustration.

**Comment by MillionsToOne** [comment]
It isn't. It's the first place a person first time using the app would look, hence it being there. After all, all of those options listed there do the same thing, add items.

**Comment by Daniel15** [comment]
Wow, this is extemely confusing. I was having the same issue when trying to import a .authpro backup to migrate over to Stratum. Import from other apps -> Authenticator Plus always complained that the password was wrong. I didn't notice that it actually wanted a **different** type of file: `authplus.db`. Using "Restore a backup" instead of "Import from other apps" worked perfectly.

**Comment by Nowaker** [comment]
Both the UX and the messaging of errors is crazy bad.

**Comment by MillionsToOne** [comment]
"Import from other apps" does exactly what it's supposed to do. Import from **OTHER** apps. Maybe reading next time what each button does before doing anything would help

**Comment by Daniel15** [comment]
When migrating from Authenticator Pro to Stratum, wouldn't you consider Authenticator Pro as an "other app? It seems reasonable to look in there.

**Comment by MillionsToOne** [comment]
> When migrating from Authenticator Pro to Stratum, wouldn't you consider Authenticator Pro as an "other app? It seems reasonable to look in there.
> 
> 
> On December 1, 2024 10:47:55 PM PST, MillionsToOne ***@***.***> wrote:
> >"Import from other apps" does exactly what it's supposed to do. Import from **OTHER** apps. Maybe reading next time what each button does before doing anything would help
> >
> >-- 
> >Reply to this email directly or view it on GitHub:
> >https://github.com/stratumauth/app/issues/1165#issuecomment-2510707295
> >You are receiving this because you commented.
> >
> >Message ID: ***@***.***>

Because Stratum is just a rebranding, not another app.

**Comment by Nowaker** [comment]
@MillionsToOne Are you really expecting import feature to be under a "+" button that's primarily used for adding new QR codes?

I suggest you give https://github.com/stratumauth/app/issues/1165#issuecomment-2241931657 a re-read because you come across as a troll, and not a productive member of an open source community.

**Comment by MillionsToOne** [comment]
> @MillionsToOne Are you really expecting import feature to be under a "+" button that's primarily used for adding new QR codes?
> 
> I suggest you give https://github.com/stratumauth/app/issues/1165#issuecomment-2241931657 a re-read because you come across as a troll, and not a productive member of an open source community.

Oh yes, I expect it to be there.

**Comment by jamie-mh** [comment]
This is somewhat related to #1169 and, since the app is no longer called "Authenticator Pro", there shouldn't be any confusion with "Authenticator Plus".
